### DISCUSSION TITLE
Request for Ideas/Feedback: Infix syntax macro

### DISCUSSION THREAD

**Discussion by wrq** [janet-lang/janet#675] [Ideas]
Scheme has these via SRFI-105:
https://srfi.schemers.org/srfi-105/

It'd be nice for occasional math-heavy definitions. I think a decent portion of Janet users are interested or perhaps even using it for gamedev, and I would love some easier way to deal with writing custom collision code, linerp stuff, anything beyond the basic xy utilities is just kind of blinding to do using regular s-expressions.

Maybe this doesn't belong in the standard library, I'm not really sure. Even still, maybe this discussion will be able to help get more ideas about this functionality and we could organize an implementation as a jpm package.

**Comment by subsetpark** [comment]
I’ll admit this has never arisen as a need for me, but: do you have any thoughts as to initial directions? Copying the SRFI directly wouldn’t work since we already use curly braces. We also don’t have reader macros, for better or worse, so I suppose it’d have to be added at the language level. 

(Maybe the discussion should be about the feasibility of adding reader macros!)

**Comment by bakpakin** [comment]
I would much prefer something like an `expr` macro that expanded things inside it with infix syntax (a la TCL). Same idea really, but no special syntax for it (no curly braces, we already use those for stuff). Probably wouldn't make it into the core library as it is not a necessity but it could certainly go into a utility library like spork.
